src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1517,23 +1588,45 @@ MulticopterPositionControl::task_main()
 				}
 
 			} else if (_control_mode.flag_control_manual_enabled
-				   && _vehicle_land_detected.landed) {
-				/* don't run controller when landed */
-				_reset_pos_sp = true;
-				_reset_alt_sp = true;
-				_do_reset_alt_pos_flag = true;
-				_mode_auto = false;
-				reset_int_z = true;
-				reset_int_xy = true;
+				   && (landed_for_sure || (_landed_locked && _acc_z_lp < 4.0f))) {
```

Comment:
this number is anyway rather strange: at hover, the _acc_z_lp is around 0, when accelerating upward, it is negative (z is positive pointing upward), and when going down the acceleration is limited anyway (I only can go down very slowly with small acceleration), I will never achieve an acceleration of 4.0f downwards. 
I also tested what the values are at landing, and they are sliglty negative (~ -0.17). Hence, this 4.0f has currently not really any effect.